### PR TITLE
Add hmftools-cuppa 1.8.1

### PR DESCRIPTION
**Pull request by scwatts**
Add recipe for hmftools-cuppa 1.8.1

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by scwatts** [comment]
@BiocondaBot please add label

File: recipes/hmftools-cuppa/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+build:
+  noarch: generic
+  number: 0
+  run_exports:
+    - {{ pin_subpackage("hmftools-cuppa", max_pin="x.x") }}
+
+requirements:
+  run:
+    - openjdk >=8
+    - r-base ==4.2.2
```
Comment (bgruening) [inline recipes/hmftools-cuppa/meta.yaml:26]: can we relax this pin please a bit

Comment (scwatts) [reply recipes/hmftools-cuppa/meta.yaml:26]: Yep, no problem

**Comment by scwatts** [comment]
The linting job hung until reaching 60 minute timeout:

```
00:45:57 BIOCONDA DEBUG get_recipes(recipes/, package='['*']'): *
00:45:57 BIOCONDA INFO Considering total of 9240 recipes.
00:45:57 GIT DEBUG Popen(['git', 'remote', 'get-url', '--all', 'origin'], cwd=/home/vsts/work/1/s, universal_newlines=False, shell=None, istream=None)
00:45:57 BIOCONDA WARNING Couldn't get the active branch name, we must be on detached HEAD
00:45:57 GIT DEBUG Popen(['git', 'merge-base', 'origin/master', 'pull/45643/merge'], cwd=/home/vsts/work/1/s, universal_newlines=False, shell=None, istream=None)
00:45:57 GIT DEBUG Popen(['git', 'cat-file', '--batch-check'], cwd=/home/vsts/work/1/s, universal_newlines=False, shell=None, istream=<valid stream>)
00:45:57 GIT DEBUG Popen(['git', 'diff-tree', 'decf1eadb865e7460332cd0bd0dd1e2e0e939deb', 'pull/45643/merge', '-r', '--abbrev=40', '--full-index', '-M', '--raw', '--no-color'], cwd=/home/vsts/work/1/s, universal_newlines=False, shell=None, istream=None)
00:45:57 GIT DEBUG Popen(['git', 'cat-file', '--batch'], cwd=/home/vsts/work/1/s, universal_newlines=False, shell=None, istream=<valid stream>)
00:45:57 GIT DEBUG Popen(['git', 'merge-base', 'origin/master', 'pull/45643/merge'], cwd=/home/vsts/work/1/s, universal_newlines=False, shell=None, istream=None)
00:45:57 BIOCONDA INFO Constraining to 1 git modified recipes (hmftools-cuppa).
00:45:57 BIOCONDA INFO Processing 1 recipes (hmftools-cuppa).

##[error]The operation was canceled.
```

Linting completes successfully within a couple of seconds when locally run for the hmftools-cuppa recipe, failure probably an intermittent problem. I'll re-launch the failed Azure job.

**Comment by scwatts** [comment]
@BiocondaBot please fetch artifacts

**Comment by scwatts** [comment]
@BiocondaBot please fetch artifacts

**Comment by scwatts** [comment]
@BiocondaBot please add label

File: recipes/hmftools-cuppa/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+build:
+  noarch: generic
+  number: 0
+  run_exports:
+    - {{ pin_subpackage("hmftools-cuppa", max_pin="x.x") }}
+
+requirements:
+  run:
+    - openjdk >=8
+    - r-base ==4.2
```
Comment (bgruening) [inline recipes/hmftools-cuppa/meta.yaml:26]: Upps, sorry I was less precise. Can we just specify r-base or mabye `>=4`

Comment (scwatts) [reply recipes/hmftools-cuppa/meta.yaml]: Done!

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks a lot!

**Pull request merged**